ipfix/test_ipfix.py

```diff
@@ -0,0 +1,161 @@
+"""
+
+import unittest
+
+from scapy.packet import Raw
+from scapy.layers.inet import IP, UDP, Ether
+from ipfix import IPFIX, Template
+
+from framework import VppTestCase, VppTestRunner
+from util import ppp
+from random import randint
+
+class TestIPFIXTemplate(VppTestCase):
+    """ IPFIX template test case """ # test names are required
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestIPFIXTemplate, cls).setUpClass()
+        cls.create_pg_interfaces(range(2))  # create pg0 and pg1
+        for i in cls.pg_interfaces:
+            i.admin_up()  # put the interface upsrc_if
+            i.config_ip4()  # configure IPv4 address on the interface
+            i.resolve_arp()  # resolve ARP, so that we know VPP MAC
+
+    def setUp(self):
+        super(TestIPFIXTemplate, self).setUp()
+        # FIXME: I can't figure out how to get the test framework to call the
+        #        IPFIX plugin's API instead of using the CLI like this
+        self.logger.info(self.vapi.ppcli("set ipfix ip collector " + self.pg0.remote_ip4))
+        self.logger.info(self.vapi.ppcli("set ipfix ip exporter " + self.pg1.remote_ip4))
+        self.logger.info(self.vapi.ppcli("ipfix flow-meter " + self.pg1.name))
+
+    def test_template(self):
+        self.logger.info(self.vapi.ppcli("set ipfix timeout template 1"))
+
+        self.pg0.enable_capture()
+        self.pg1.enable_capture()
+        self.pg_start()
+        # no packets on pg1 since we didn't inject any packets
+        self.pg1.assert_nothing_captured()
+        # expect 3 template pkt, 1/sec
+        capture = self.pg0.get_capture(timeout = 3, expected_count = 3)
+        self.verify_template(self.pg0, self.pg1, capture)
+
+    def verify_template(self, collector_if, exporter_if, capture):
+        for packet in capture:
+            try:
+                ip = packet[IP]
+                udp = packet[UDP]
+                ipfix = packet[IPFIX]
+                self.assert_equal(ip.src, exporter_if.remote_ip4,
+                                  "exporter ip")
+                self.assert_equal(ip.dst, collector_if.remote_ip4,
+                                  "collector ip")
+                self.assert_equal(udp.dport, 4739)
+                self.assert_equal(ipfix.version, 10)
+                # check that there's an IPFIX template
+                self.assert_equal(ipfix[Template].templateID, 256)
+            except IndexError:
+                self.logger.error(ppp("Invalid packet:", packet))
+                raise
+            except AssertionError:
+                self.logger.error(ppp("Unexpected packet:", packet))
+                raise
+
+class TestIPFIXData(VppTestCase):
+    """ IPFIX data test case """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestIPFIXData, cls).setUpClass()
+        cls.create_pg_interfaces(range(2))
+        for i in cls.pg_interfaces:
+            i.admin_up()
+            i.config_ip4()
+            i.resolve_arp()
+
```

Comment:
Another small nit, I'd have probably created a `AbstractIPFIXTestCase` class or something since the `setUpClass` and `setUp` methods seem to be shard amongst them (as well as some other stuff I mention below).